### ISSUE TITLE
How to make opencensus transparency to user application?

### ISSUE THREAD

**Issue by a651694192** [census-instrumentation/opencensus-java#1920]
I wonder if we can make opencensus transparency  to users, which means users don't need to change their application to collect the telemetry  just like spring cloud seluth do.
As I'm new to opencensus, I don't know if we already enable this because I think this is a fundamental requirement  for the distributed tracing.

**Comment by songy23** [comment]
You'll be able to get telemetry out-of-box if you're using one of the frameworks that's already instrumented with OpenCensus. See https://opencensus.io/integrations/ and https://opencensus.io/guides/. For Spring integration, we're actively working on it.

**Comment by a651694192** [comment]
> You'll be able to get telemetry out-of-box if you're using one of the frameworks that's already instrumented with OpenCensus. See https://opencensus.io/integrations/ and https://opencensus.io/guides/. For Spring integration, we're actively working on it.

But it seems even integrated framework require code inject/code change for user application, for example in your http example https://opencensus.io/guides/http/java/jetty/client/

**Comment by songy23** [comment]
That's correct. Collecting and exporting trace and metrics could be expensive (both in terms of overhead to applications, and money cost if you're using commercial APMs). Thus we want users to have full control over it, e.g what kind of metrics/trace to collect, how frequently telemetry is exported, which APM backend to use, etc.

Instead of doing explicit instrumentation in the application, you can consider using a JVM agent that manipulates Java bytecode as an alternative, for example [inspectIT Ocelot](https://github.com/inspectIT/inspectit-ocelot). /cc @JonasKunz @AlexanderWert for guidance here.
